### PR TITLE
Improve Countdown Block 

### LINKED ISSUE #825: Countdown - enhancements
- [ ] More predefined styles, like we[ initially discussed](https://github.com/Codeinwp/gutenberg-blocks/issues/737#event-5195751400). Round, filled, etc
- [x] Option to have it Responsive. For example I want to counter to be on a full screen page, but at the same time look on Mobile. Some Typography responsive settings may be needed. 
For example in the end, it should be possible to build something like this:
image
Right now something like this wouldn’t be possible, as the Time Value Font Size is limited to 64px.

- [ ] Create a new tab for Style and keep in Settings only things related to settings.
- [x] Separate Number Typography controls from the Label Typography controls. Each of them should be configurable independently. And have the usual typography options, not just the font-size.
- [x] Add relative units for Font-size
- [x] Add text/content alignment options. This will align the text inside each column, Left, Center, Right:
image

### Potential issues
- [ ] Every-time when the page loads, the timer is 0 for a second, and then the numbers are loaded. Maybe could do something about this using the available lifecycles/cache, which won’t reset the dates at each refresh.

### LINKED ISSUE #1020: Add logged-in user meta condition to Block Conditions
### Description:
This was suggested by a user at WordCamp Europe to have a condition to show blocks based on the logged-in user's meta condition. This can be paired up with different plugins that add user meta and then show it based on that, ie based on their level, role or amount of points they have.

The code is already there for it, just need to tweak it for logged-in user.

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes #825.

### Summary

@JohnPixle @mghenciu this is a small rework for Countdown. The inspector was rearranged and the width of the box has been transferred to the container (now it should behave like the one from Elementor). It should help with the WP Maintenance design.

#### ⚠️ Container behavior changed

- The container width will use the following formula for computing the width with the old values: `--container-width: clamp(400px, var(--width) * 4, 100%);`. Once the block is updated, the block will be the new system. Thus, the old value will be no longer used but kept in case the user wants to return to a previous version without destroying the layout.

### Screenshots 

----

### Test instructions

---- 

### Checklist before the final review

- [x] Visual elements are not affected by independent changes.
- [x] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [x] It loads additional script in frontend only if it is required.
- [x] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [x] In case of deprecation, old blocks are safely migrated.
- [ ] It is usable in Widgets and FSE.

### CODE REVIEW

**Comment by JohnPixle** [comment]
@Soare-Robert-Daniel Thanks! Can you please attach the corresponding zip at the original comment? I will take a look asap.

@HardeepAsrani We have these issues open for the countdown:

https://github.com/Codeinwp/otter-blocks/issues/912
https://github.com/Codeinwp/otter-blocks/issues/825
https://github.com/Codeinwp/otter-blocks/issues/804
https://github.com/Codeinwp/otter-blocks/issues/976

Feel free to manage them as you see fit, or let me know if I can help.

**Comment by HardeepAsrani** [comment]
Thanks @JohnPixle,

@Soare-Robert-Daniel Please fix the JS errors so the zip can be generated and mention the issues which this particular PR solves so we can keep track of fixed items.

**Comment by Soare-Robert-Daniel** [comment]
@JohnPixle, sorry for labeling the wrong issue. This is related to #825

#976 #804 was related to the FSE issue that made the CSS unusable in iFrame.

#912 will require more investigation since is a ghost bug.

**Comment by JohnPixle** [comment]
Hey @Soare-Robert-Daniel thanks for the update. I tested the **Production** version of Otter, from the comment above.

A few things I spotted:

- [ ] The **time value font size affects both the time value and the label.** The control for the actual label does not have any effect.

Screenshot 2022-07-11 at 16 03 09

- [ ] **Separators.** The separators are vertically top aligned, but they should be centered aligned always. Additionally, the separator color is affected by the value color, so if value color is white, then the separator is not visible.

Screenshot 2022-07-11 at 16 09 22

- [ ] Width panel is a bit messed up. The width should be in %, now it seems to be in pixels. As a result (I assume) the boxes go out of the container. 

Screenshot 2022-07-11 at 16 20 00

- [ ] There are two `//` characters in the UI (under the width control), I think they must be there by mistake.

- [ ] **Alignment does not seem to work properly.** Is the alignment control supposed to affect the alignment of the time value and label? IF so, it does not seem to do the trick, as it seems to apply on the actual boxes. 

- [ ] Inheritance of text sizes on mobile viewports. **The font size that is defined at the desktop view, is not being inherited to the mobile viewports** (pad / mobile). The user needs to define the font sizes again for these viewports.

Screenshot 2022-07-11 at 16 50 57


Let me know if this helps and if you need any clarifications!

**Comment by Soare-Robert-Daniel** [comment]
@JohnPixle can you clarify the alignment option? I understand that if you have a small container, you want to align it.

Like this:
image

**Comment by Soare-Robert-Daniel** [comment]
@JohnPixle can you take a look at the current state when you have time?

**Comment by Soare-Robert-Daniel** [comment]
@JohnPixle @mghenciu I prepared an instance to test out the changes:

Page: https://accidentaldock.s4-tastewp.com/2022/07/13/countdown/

 Admin area URL: https://accidentaldock.s4-tastewp.com/wp-admin 
 Username: robertino 
 Password: 83fasUUrLl4

**Comment by Soare-Robert-Daniel** [comment]
@mghenciu I think we can approach the time display when is loading as Elementor does by making it visible after the script is loading. In the past, we decide to show the number so that the people who used AMP can see that is something there. I don't think there is a case now.

What do you think?

**Comment by mghenciu** [comment]
Sorry for the late reply, Robert.
First of all, congrats -> You've done a really nice job  🚀 🦦 
image

___

Just want to mention a few minor things:

### Sidebar UI
The sidebar Looks great, 

- [ ] I would rename the Sizing tab to **Dimensions & Spacing** 
___
- [ ] For the border options, I think it should start with None, and then the other options:
image
___
- [ ] in the Color Panel (and everywhere else), we should follow the same logical structure, exactly how the Countdown is structured. So in this case it should be this one 
image

- [ ] I am not sure if we need the Padding Control right now, as we don't Box Min Width/Height. **Maybe we can hide it for now**, and add it later, with the Sidebar redesign.
___

- [ ] How difficult it would be to add **Font Weight options to the Time value Text**? If it takes a lot of time, we can leave it for later. But for sure we need this at some point.

___
### Mobile

- [ ] on mobile the Height of the container is somehow inherited, even if the box is smaller:
https://user-images.githubusercontent.com/52494172/178951875-4fa32138-4b72-4b5b-94f0-b69fe80ae6d6.mov

**Comment by mghenciu** [comment]
> @mghenciu I think we can approach the time display when is loading as Elementor does by making it visible after the script is loading. In the past, we decide to show the number so that the people who used AMP can see that is something there. I don't think there is a case now.
> 
> What do you think?

If I understand it correctly, it's about showing the Number/something directly when the page loads.
Right now I think it's a bit confusing, because in the Initial Phase, our countdown show **0 to all values** - and after a few milliseconds it shows the actual value.

**Ideally I think there should be some preload state, either skeleton loading for the numbers or a slower render of the Block, until it has enough time to get the actual Time value.** 
For sure showing 0 is not the best scenario.

**Comment by mghenciu** [comment]
✅ The skeleton animation looks good, Robert. **I would just bind its color to the Text Color**, with a 0.05 opacity. This way it will look good both on Light & Dark Mode (being a dynamic color), and we don't need to worry about how it will look on diferent colors. And probably it's best to apply the same effect on the Labels (Day, Minute, Seconds, etc).
✅ Font weight also looks good


For the Height, I not 100% we need it. Probably we'll review this more when we'll build the Coming Soon pages.
Same thing for the **stacking on Mobile**, a safe bet would be to make this a Toggle, and not the default option. As I haven't seen Countdowns stacked on Mobile, usually those follow the same structure as on desktop.

**Comment by Soare-Robert-Daniel** [comment]
> ✅ The skeleton animation looks good, Robert. **I would just bind its color to the Text Color**, with a 0.05 opacity. This way it will look good both on Light & Dark Mode (being a dynamic color), and we don't need to worry about how it will look on diferent colors. And probably it's best to apply the same effect on the Labels (Day, Minute, Seconds, etc). ✅ Font weight also looks good

Can't do that. Applying an alpha channel to an arbitrary value is not possible dynamically. [More about it](https://stackoverflow.com/questions/40010597/how-do-i-apply-opacity-to-a-css-color-variable).

As a reference, the color is saved as HEX values or var values like `var(--neve-pc)'. We are very limited on this side.

We need a color that fits on Light & Dark Mode or thinking about another approach.

**Comment by JohnPixle** [comment]
> Same thing for the stacking on Mobile, a safe bet would be to make this a Toggle, and not the default option. As I haven't seen Countdowns stacked on Mobile, usually those follow the same structure as on desktop.

You are right @mghenciu about the stacking on mobile. Let's leave it as it is for now, we can re-visit later if needed.

**Comment by Soare-Robert-Daniel** [comment]
@mghenciu @JohnPixle you can how breaking on mobile work at 

```
 Page: https://ajarrod.s4-tastewp.com/?p=6&preview=true
 Admin area URL: https://ajarrod.s4-tastewp.com/wp-admin 
 Username: robertino 
 Password: mCPoy8-jmyU 
```

Check the Dimensions and Spacing Tab

**Comment by mghenciu** [comment]
@Soare-Robert-Daniel , so using rgba(var(--neve-text-color), 0.05) or HSL in the same way, won't work or will brake the scss processor?

___

Just to confirm we are on the same page about the Stacking on Mobile: **it should be optional (not the default)**, meaning that it will work only when the Toggle is on.
Right now I've added a Countdown, Toggled On and now Off, but it keeps stacking on Mobile:
https://ajarrod.s4-tastewp.com/countdown-2/

**Comment by JohnPixle** [comment]
Hey @Soare-Robert-Daniel thanks for the update. Regarding the "stack on mobile" , I spotted some issues as shown in this [2-min video](https://www.loom.com/share/3632c5e99c9644208297865172d8d8ca), but overall I would suggest that **we remove the toggle completely, and revert to the earlier version** where there was not stacking.

- [ ] Remove "stack on mobile" toggle and functionality.

Mihai's suggestion was correct, it's not common practice for countdowns to be stacked on mobile, usually they are on a single row on mobile. Sorry for the back and forth!

**Comment by Soare-Robert-Daniel** [comment]
> @Soare-Robert-Daniel , so using rgba(var(--neve-text-color), 0.05) or HSL in the same way, won't work or will brake the scss processor?

it just does not work. It is the same problem that we had when we wanted to add opacity to the shadow and at the same time use color var to be dynamic.

**Comment by Soare-Robert-Daniel** [comment]
@JohnPixle @mghenciu do you have any requests left or questions about some implementation?

File: src/blocks/blocks/countdown/deprecated.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		const borderRadiusBox = 'linked' === attributes.borderRadiusType ?
+			{ left: optionalUnit( attributes.borderRadius ), right: optionalUnit( attributes.borderRadius ), bottom: optionalUnit( attributes.borderRadius ), top: optionalUnit( attributes.borderRadius ) } :
+			{ left: optionalUnit( attributes.borderRadiusBottomLeft ), right: optionalUnit( attributes.borderRadiusTopRight ), bottom: optionalUnit( attributes.borderRadiusBottomRight ), top: optionalUnit( attributes.borderRadiusTopLeft ) };
+		oldAttributes.borderRadiusBox = borderRadiusBox;
+
+		return omit( oldAttributes, [ 'borderRadiusBottomLeft', 'borderRadiusTopRight', 'borderRadiusBottomRight', 'borderRadiusTopLeft', 'borderRadiusType' ]);
+	},
+
+	isEligible: ({ borderRadius }) => borderRadius && 'number' === typeof borderRadius,
+
```
Comment (Soare-Robert-Daniel) [inline src/blocks/blocks/countdown/deprecated.js:62]: @HardeepAsrani can you tell if this part is ok? It seams that `borderRadiusBox` is not computed

Comment (HardeepAsrani) [reply src/blocks/blocks/countdown/deprecated.js:62]: The only way to find out is to test really. You can just copy/paste old blocks from another editor and will find, really can't say much from code.

**Comment by HardeepAsrani** [comment]
Given that Robert is deprecating the Countdown Block, I'll recommend you all to tell him of every possible change that you think needs to be done, so we don't have to deprecate it again in near future. 😄

**Comment by JohnPixle** [comment]
I cannot think of anything else at this point, we can revisit at a later stage.

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani, you can take a look. I need feedback on the loading animation. This is how it looks now.

https://user-images.githubusercontent.com/17597852/179746433-59d36ec2-e892-4902-b4a5-234cdc2de875.mp4

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Tested and here are my findings:

- [ ] Previously created blocks will not look the same after updating to this PR
Before the PR: https://vertis.d.pr/6qzj77
With the PR: https://vertis.d.pr/DubxC9
And if i just click on **edit page** and hit the **Update** button without making any change, it will look like this https://vertis.d.pr/8XHqXb
- [ ] For previously created blocks, some editor controls will look like this https://vertis.d.pr/ozZuWk
- [ ] The max value for Height is really big (2400) https://vertis.d.pr/tgyKGQ

**Comment by JohnPixle** [comment]
Ooook, thanks so much for the detailed report @irinelenache. 

We need to take some decisions I believe. There are some blocks that have fundamental issues that need to be addressed before we dive deeper into layouts and patterns. On the other hand, making changes breaks the appearance of the block at user sites. Need to get the best out of this situation.

How logical would it be to release major block reworks as new blocks @HardeepAsrani ? I think it would take half the time it would take us to work on fixing the existing ones without breaking the sites.

- Existing blocks are not affected at users sites
- Users are able to insert only the new versions of the blocks.

**Comment by Soare-Robert-Daniel** [comment]
@mghenciu @JohnPixle @HardeepAsrani as a note, when the client will enter to the page, the previous width settings will not be used, since now we take in the consideration the container width and not individual width. 

The old value will remain, if someone change back to a previous version, the block will not be affected.

For the border radius, the individual values is dropped in the favour of a new system.

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Tested again and the block works fine. We should keep in mind that user should be somehow notified that if they used the block before the update, it will not look the same now and they have to set it again

If @mghenciu & @JohnPixle approve, we can move this to **Ready to merge** 👍

**Comment by HardeepAsrani** [comment]
It has some breaking small inconveniences for users coming from the previous version, but they seem to be small and not affect the design much + they can always just update the page to fix it. Plus, I expect Countdown isn't the most used block to cause much issues so we can afford an update at this point in its life.

Let me know @mghenciu & @JohnPixle if you're satisfied with all the changes or anything else needs to be done.

**Comment by mghenciu** [comment]
> Let me know @mghenciu & @JohnPixle if you're satisfied with all the changes or anything else needs to be done.

Looks good! As mentioned above.

However it's best to approach things like this differently in future, meaning that if we release a block in the wild, it should have a really good base, and should't affect past versions a lot. For the Countdown block, even if not used by a lot of users, on release it had a lot of improvements missing.

**Comment by HardeepAsrani** [comment]
@mghenciu Totally agree.

**Pull request merged**